src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -135,6 +152,16 @@ module DotNet {
      * @param argsJson JSON representation of arguments to pass to the method.
      */
     beginInvokeDotNetFromJS(callId: number, assemblyName: string | null, methodIdentifier: string, dotNetObjectId: number | null, argsJson: string): void;
+
+    /**
+     * Invoked by the runtime to begin an asynchronous call to a .NET method.
+     *
+     * @param callId A value identifying the asynchronous operation. This value should be passed back in a later call from .NET to JS.
+     * @param succeded Whether the operation succeeded or not.
+     * @param resultOrError The result or the error from the async operation.
+     * @param argReplacer A function that replaces [DotNetObject] references with the appropriate value during serialization.
+     */
+    endInvokeDotNetFromJS?(callId: number, succeeded: boolean, resultOrError: any, argReplacer: (key: string, value: any) => any): void;
```

Comment:
Should this actually be called `endInvokeJSFromDotNet`? I can't understand the meaning of it otherwise. The default implementation works by making a call into .NET, which wouldn't make sense if we were completing a call that originated in JS (as the completion would be going to JS, not to .NET).

Reply:
Also TBH I'd prefer for this to be a required property on the interface, and not to have a default implementation here. The implementation can be provided by the platform - we already have a way to plug in an implementation for WebAssembly separately from server-side Blazor. I don't see why we'd have a default implementation for this but not for `beginInvokeDotNetFromJS`.

So altogether perhaps this should be declared as:

```suggestion
    endInvokeJSFromDotNet(callId: number, succeeded: boolean, result: string): void;
```

... where the platform's job is simply to pass through the `callId` and `result` string to `DotNetDispatcher.EndInvoke` on the .NET side.

Reply:
Ok, lets do that.

Reply:
done